Casks/confluent-hub-client.rb

```diff
@@ -0,0 +1,23 @@
+cask 'confluent-hub-client' do
+  version '5.1.0-SNAPSHOT'
+  sha256 'a0c008f95cd3cd01b3a24729a27307ac5d01990cd9b73b185ccfb0728677af80'
+  url "https://s3-us-west-2.amazonaws.com/confluent-hub-client/ConfluentHubClient-#{version}.pkg"
+  name 'Confluent Hub Client'
+  homepage 'https://www.confluent.io/hub/'
+  pkg "ConfluentHubClient-#{version}.pkg"
+
+   postflight do
+    system_command '/bin/ln',
+                   args: ['-nsf', '--', "/Applications/ConfluentHubClient.app/Contents/bin/confluent-hub", '/usr/local/bin/confluent-hub'],
+                   sudo: true
+    system_command '/bin/ln',
```

Comment:
this crashed for me because I don't have `/usr/local/share/java` as a directory. 

```
20:03 $ brew cask install confluent-hub-client
==> Satisfying dependencies
==> Downloading https://s3-us-west-2.amazonaws.com/confluent-hub-client/ConfluentHubClient-5.1.0-SNAPSHOT.pkg
######################################################################## 100.0%
==> Verifying checksum for Cask confluent-hub-client
==> Installing Cask confluent-hub-client
==> Running installer for confluent-hub-client; your password may be necessary.
==> Package installers may write to any location; options such as --appdir are ignored.
Password:
installer: Package name is ConfluentHubClient
installer: Installing at base path /
installer: The install was successful.
ln: /usr/local/share/java/confluent-hub-client: No such file or directory
==> Purging files for version 5.1.0-SNAPSHOT of Cask confluent-hub-client
Error: Failure while executing; `/usr/bin/sudo -E -- /bin/ln -nsf -- /Applications/ConfluentHubClient.app/Contents/share/java/confluent-hub-client /usr/local/share/java/confluent-hub-client` exited with 1. Here's the output:
ln: /usr/local/share/java/confluent-hub-client: No such file or directory

```

it also worried me a bit to see the need for `sudo`. One of the nice things about brew is that it seems to rarely require that. Is this because `casks` usually don't have CLI interfaces? Do other casks do something like this if they need to provide convenient access to CLIs?

Reply:
I've noticed, that most of `brew cask` formulas are Desktop applications, so those can reside  under `/Applications/<AppName>` without need to create symlinks elsewhere(`/usr/local/bin/` and ` /usr/local/share/java` in our case, that's why we need a `sudo` access). Those formulas, which are not a pure Desktop application, they create symlinks outside of `/Applications`, for example Java cask - https://github.com/Homebrew/homebrew-cask/blob/master/Casks/java.rb.
If we want to have the client fully reside under `/Applications/ConfuentHubClient.app` we can create an executable file, which would open a Terminal with exported environment variables. But in this case, the client would be not accessible from a regular terminal session.

Reply:
I don't think we want it to open a terminal, that wouldn't be the right user experience. I think we're just in slightly weird territory here wrt the type of app (a proprietary CLI). symlinking could be ok, just unfortunate to need sudo access.

although i thought brew set up /usr/local/bin to not require root so that it could symlink to binaries. /usr/local/share/java might require it though since it is non-standard on mac.

Reply:
@ewencp I did some more research and found some casks, which are similar to ours(android-sdk). It looks like we can use regular archive, not even DMG or PKG and reside within `/usr/local/Caskroom/confluent-hub-client`, linking `confluent-hub` to /usr/local/bin/confluent-hub. With corresponding updates of binary script here https://github.com/confluentinc/hub-client/pull/31/files, it works.